source/globalCommands.py

```diff
@@ -4009,6 +4009,18 @@ def _enableScreenCurtain(doEnable: bool = True):
 			else:
 				_enableScreenCurtain()
 
+	@script(
+		description=_(
+			# Translators: Describes a command.
+			"Toggles through paragraph navigation styles",
+		),

```

Comment:
This script has no category and ends up in the misc category. Please add one. I think `SCRCAT_SYSTEMCARET` would be OK.

```suggestion
		),
		category=SCRCAT_SYSTEMCARET,
```

Reply:
@CyrilleB79 Your Notepad example in multi line break style is correct. You'll probably disagree, but you need to accept that you are no longer in the second paragraph when at the fourth line (a blank line.) Behavior is the same no matter how many blank lines you put at the end. Believe me, this is the behavior you want.

Reply:
> @CyrilleB79 Your Notepad example in multi line break style is correct. You'll probably disagree, but you need to accept that you are no longer in the second paragraph when at the fourth line (a blank line.) Behavior is the same no matter how many blank lines you put at the end. Believe me, this is the behavior you want.

After reading your words, I have made some more tests. And I finally agree that the current implementation makes sense. The confusion was coming from the following point. When in multi-line break style, if the cursor is on an empty line, we should not consider that we are in a paragraph but between two paragraphs.
I have also made tests with Jaws, which uses multi-line break as paragraph separator. The UX with NVDA multi-line break paragraph style is quite similar to Jaws, except a bug in Jaws witch is not present in your implementation for NVDA.

In conclusion, do not take into account my remark "When the cursor is not at the beginning of the paragraph".